<launch>
    <include file="$(find rcv)/launch/shared.launch" pass_all_args="true" />

    <node
        name="$(eval vehicle_id + '_distance_node')"
        pkg="common"
        type="distance.py"
        respawn="true"
    />
    <node
        name="$(eval vehicle_id + '_object_follower_node')"
        pkg="common"
        type="object_follower.py"
        respawn="true"
    />
    <node
        name="$(eval vehicle_id + '_pixy2_node')"
        pkg="pixy2_node"
        type="pixy2_node"
        respawn="true"
    >
        <param name="rate" value="60.0" />
        <param name="use_servos" value="false" />
        <param name="enable_lamp" value="false" />
    </node>

    <!-- Object follower -->
    <param name="VEHICLE_STOP_IF_NO_OBJECT_VISIBLE" type="bool" value="true" />
    <param name="OBJECT_FOLLOWER_BLOCKS_PER_UPDATE" type="int" value="1" />
    <param name="OBJECT_FOLLOWER_PID_KP" type="double" value="1" />
    <param name="OBJECT_FOLLOWER_PID_KI" type="double" value="0.02" />
    <param name="OBJECT_FOLLOWER_PID_KD" type="double" value="0.05" />
    <param name="OBJECT_FOLLOWER_PID_SETPOINT" type="double" value="0" />
    <param name="OBJECT_FOLLOWER_PID_MIN" type="double" value="-200" />
    <param name="OBJECT_FOLLOWER_PID_MAX" type="double" value="200" />

    <!-- Minimum and maximum velocity allowed. -->
    <param name="VELOCITY_MIN" type="int" value="0" />
    <param name="VELOCITY_MAX" type="int" value="8" />

    <!-- GPIOs BOARD -->
    <param name="ULTRASONIC_TRIGGER_1" type="int" value="13" />
    <param name="ULTRASONIC_ECHO_1" type="int" value="11" />
    <param name="ULTRASONIC_TRIGGER" type="int" value="29" />
    <param name="ULTRASONIC_ECHO" type="int" value="31" />
    <param name="ULTRASONIC_TRIGGER_3" type="int" value="16" />
    <param name="ULTRASONIC_ECHO_3" type="int" value="15" />

    <!-- Publisher periods -->
    <param name="DISTANCE_PUBLISH_PERIOD" type="double" value="0.01" />
    <param name="OBJECT_FOLLOWER_PERIOD" type="double" value="0.0167" />

    <!-- Ultrasonic node -->
    <param name="MINIMUM_DISTANCE_THRESHOLD" type="int" value="10" />
    <param name="ULTRASONIC_SAMPLES_PER_PUBLISH" type="int" value="5" />
    <param name="ULTRASONIC_PING_PULSE_LENGTH" type="double" value="0.00001" />
    <param name="ULTRASONIC_TIMEOUT_LENGTH" type="double" value="0.05" />
    <param name="ULTRASONIC_MIN_RANGE" type="int" value="3" />
    <param name="ULTRASONIC_MAX_RANGE" type="int" value="400" />
    <param name="ULTRASONIC_FOV" type="double" value="0.26179938779915" />
</launch>
